src/stdlib_stats_median.fypp

```diff
@@ -0,0 +1,217 @@
+#:include "common.fypp"
+#:set RANKS = range(1, MAXRANK + 1)
+#:set IR_KINDS_TYPES_OUTPUT = list(zip(REAL_KINDS, REAL_TYPES, REAL_KINDS)) + list(zip(INT_KINDS,INT_TYPES, ['dp']*len(INT_KINDS)))
+
+
+submodule (stdlib_stats) stdlib_stats_median
+
+  use, intrinsic:: ieee_arithmetic, only: ieee_value, ieee_quiet_nan
+  use stdlib_error, only: error_stop
+  use stdlib_optval, only: optval
```

Comment:
Here I suggest to add a comment explaining why you are using `ord_sort` rather than `sort`, e.g. 
```
! Use "ord_sort" rather than "sort" because the former can be much faster for arrays that are already partly sorted. While it is slightly slower for random arrays, ord_sort seems a better overall choice. 
```

Reply:
Thank you. I copied-pasted your comment.